src/main/scala/treadle/stage/phases/PrepareAst.scala

```diff
@@ -17,24 +27,58 @@ trait TreadlePhase extends Phase {
     annotationSeq.flatMap {
       case FirrtlCircuitAnnotation(circuit) =>
         val state = CircuitState(circuit, HighForm, annotationSeq)
-        val newState = transforms.foldLeft(state) { case (prevState, transform) => transform.runTransform(prevState) }
+        val newState = transforms.foldLeft(state) {
+          case (prevState, transform) => transform.runTransform(prevState)
+        }
         Some(TreadleCircuitStateAnnotation(newState))
       case other =>
         Some(other)
     }
   }
 }
 
+/** This provides a series of transforms that
+  * seem to be important to Treadle functionality
+  * This was based on [[firrtl.LowFirrtlOptimization]] but that
+  * has includes [[passes.memlib.VerilogMemDelays]] which can cause combinational loops
+  * for some firrtl files
+  */
+class TreadleLowFirrtlOptimization extends SeqTransform {
+  def inputForm: CircuitForm = LowForm
+  def outputForm: CircuitForm = LowForm
+
+  def transforms: Seq[Transform] = Seq(
+    passes.RemoveValidIf,
+    new firrtl.transforms.ConstantPropagation,
+    passes.PadWidths,
+    new firrtl.transforms.ConstantPropagation,
+    passes.Legalize,
+    new firrtl.transforms.ConstantPropagation,
+    passes.SplitExpressions,
```

Comment:
I wonder how getting rid of `SplitExpressions`  would affect performance.
@chick have you every tried if generating fewer intermediate wires improved performance?

Reply:
I have tried it and it did not seem to make a big difference in execution. It might help more on large circuits, but I don't have that many good examples. riscv-mini, which I am working on to make it use Treadle might provide better ones.

Reply:
A DRY-er approach would be to extend the existing `LowFirrtlOptimization` but override the transforms. I took this approach in rfuzz: https://github.com/ekiwi/rfuzz/blob/818ea683c0479245415819eda4baf82c189b65e7/instrumentation/src/main/scala/hardwareafl/firrtltransforms/CustomVerilogCompiler.scala#L117

Reply:
You'll notice I have a custom VerilogMemDelays (lol) but it was a different issue where I needed a bugfix before it was available.